theseus/theseus_layer.py

```diff
@@ -94,3 +120,157 @@ def device(self) -> torch.device:
     @property
     def dtype(self) -> torch.dtype:
         return self.objective.dtype
+
+
+def _forward(objective, optimizer, optimizer_kwargs, input_data):
+    objective.update(input_data)
```

Comment:
I'm not sure if we should use `Objective.update()` or `Objective.setup()`. In `experimental`, `Objective.update()` assumes no changes of the batch size.